nservicebus/sagas/concurrency.md

```diff
@@ -1,48 +1,50 @@
 ---
-title: Saga Concurrency
-summary: NServiceBus gives ACID semantics, using underlying storage so only one worker thread hitting a saga instance can commit.
+title: Saga concurrency
+summary: NServiceBus ensures consistency between saga state and messaging.
 component: Core
 tags:
 - Saga
 redirects:
 - nservicebus/nservicebus-sagas-and-concurrency
-reviewed: 2019-02-27
+reviewed: 2019-09-24
 related:
 - persistence/nhibernate/saga-concurrency
 - persistence/ravendb/saga-concurrency
 - persistence/sql/saga-concurrency
 ---
 
-If the endpoint is configured to allow concurrent processing of messages (default) or is scaled out, it is possible that multiple messages will hit the same saga instance simultaneously. To give ACID semantics in this situation, NServiceBus uses the underlying storage to produce consistent behavior, only allowing one of messages to complete. NServiceBus handles most of this automatically but there are some caveats.
+An [endpoint](/nservicebus/concept-overview.md#endpoint) may be configured to allow [concurrent processing of messages](/nservicebus/operations/tuning.md#tuning-concurrency). On a machine with more than one logical CPU, concurrent processing is enabled by default. An endpoint may also be [scaled out](/nservicebus/architecture/scaling.md#scaling-out-to-multiple-nodes) to multiple nodes. In these scenarios, multiple messages may be handled simultaneously by a single logical saga. To ensure consistency between saga state and messages, NServiceBus ensures that only one message handling thread creates, updates, or deletes saga state in the same transaction as receiving the message and sending new messages.
```

Comment:
> On a machine with more than one logical CPU, concurrent processing is enabled by default.

I'm removing this sentence entirely. The details of default settings, etc. are in the linked article.

> > NServiceBus ensures that only one message handling thread creates, updates, or deletes saga state in the same transaction as receiving the message and sending new messages.

> We really shouldn't be talking about threads here since we don't manage threads at all, but instead are looking at concurrent tasks.

I'll rework this (and the rest of the document) to remove references to threads.

> Also, this sentence seems to ignore the the scaled out scenario entirely. Perhaps this paragraph needs to be split to not try and combine both scenarios? They really aren't the same thing an aren't handled the same way.

What is different about the two scenarios? As I understand it, we still guarantee the same level of transactionality, whether it's between two handlers in the same process, or two handlers in different processes.